### PR TITLE
feat: manual cherry merge PR #633 to `release-1.0`

### PR DESCRIPTION
**Pull request by Kuri-su**
### What is changed and how it works?

manual cherry merge PR #633 to `release-1.0`

### CODE REVIEW

**Comment by Kuri-su** [comment]
@csuzhangxc ok, i'm done , PTAL :smile:

File: dm/master/server_test.go [hunk 1]
```diff
@@ -405,13 +405,13 @@ func (t *testMaster) TestCheckTask(c *check.C) {
 	c.Assert(err, check.IsNil)
 	c.Assert(resp.Result, check.IsFalse)
 
-	// simulate invalid password returned from dm-workers, so cfg.SubTaskConfigs will fail
+	// simulate invalid password returned from scheduler, but config was supported plaintext mysql password, so cfg.SubTaskConfigs will success
```
Comment (csuzhangxc) [inline dm/master/server_test.go:408]: should still be `dm-workers` instead of `scheduler` in release-1.0.

File: dm/worker/config_test.go [hunk 2]
```diff
@@ -64,17 +64,14 @@ func (t *testServer) TestConfig(c *C) {
...
 	clone1.ServerID = 101
-	clone2, err := cfg.DecryptPassword()
-	c.Assert(err, IsNil)
+	clone2 := cfg.DecryptPassword()
 	c.Assert(clone2, DeepEquals, clone1)
 
 	cfg.From.Password = "xxx"
-	_, err = cfg.DecryptPassword()
-	c.Assert(err, NotNil)
+	cfg.DecryptPassword()
```
Comment (csuzhangxc) [inline dm/worker/config_test.go:71]: It seems we can remove these two lines (L70 and L71).

**Pull request merged**